util/logger/src/lib.rs

```diff
@@ -157,14 +157,46 @@ fn enable_ansi_support() {
 #[cfg(not(target_os = "windows"))]
 fn enable_ansi_support() {}
 
+// Parse crate name leniently in logger filter: convert "-" to "_".
+fn convert_compatible_crate_name(filter: &str) -> String {
+    let mut filter_new = Vec::new();
+    for part in filter.split(',') {
+        let mut conf = part.split('=');
+        if let Some(ref first) = conf.next() {
+            if let Some(ref second) = conf.next() {
+                if first.contains('-') {
+                    let mut conf_new = first.replace("-", "_");
+                    conf_new.push_str("=");
+                    conf_new.push_str(second);
+                    for next in conf {
+                        conf_new.push_str(next);
+                    }
+                    filter_new.push(conf_new);
+                    continue;
+                }
+            }
+        }
+        filter_new.push(part.to_owned());
+    }
+    filter_new.join(",")
+}
+
+#[test]
+fn test_convert_compatible_crate_name() {
+    let filter = "info,a-b=trace,c-d_e-f=warn/*[0-9],g-h-i=debug/[^-]*,jkl=trace";
```

Comment:
There is only one regex filter for all modules, the correct format should be `info,a-b=trace,c-d=warn,g-h-i=debug/one-single-regex-for-all`, that means you may simplified the `convert_compatible_crate_name` fn to `filter.split('/').next().replace('-', '_') + next()`